### DISCUSSION TITLE
What is the simplest way to maintain multiple "accounts" per client?

### DISCUSSION THREAD

**Discussion by Nuhvi** [maxcountryman/axum-login#226] [Q&A]
Just like Twitter/X or Google/Gmail, the same client may want to keep the credentials of multiple accounts for users to switch between them or use the appropriate account when needed.

My understanding is that the options are:
1. Redesign the API to make every user a subdomain, and event then tower_session doesn't allow `with_domain()` per session.
2. Have a separate session for `auth_multi` cookie like Twitter, keeping track of all the accounts a client is authorized to.
3. Change the Session on the backend to contain multiple accounts, and the user explicitly specify which account it wants to use in the path or query params (or have another field in the session for active account).

None of these options seem to be doable using axum-login, is that accurate?

Thanks and great work on this crate.

**Comment by maxcountryman** [comment]
I'm not really following when you say none of these options are doable.

Firstly tower-sessions does support setting a domain.

Secondly how you associate accounts with a user is not something axum-login has an opinion about.

For example, consider that an application can have a notion of both a user and an account. A user may be associated to any number of accounts. Then "switching" accounts is just a matter of flipping a bit to indicate which of the valid accounts is currently active; this could be stored in the session as well.
